### ISSUE TITLE
last-by table to stream table

### ISSUE THREAD

**Issue by devinrsmith** [deephaven/deephaven-core#2407]
It would be nice to be able to turn a last-by table into a stream table. Specifically, I think it could be a useful construction for clients to use - they would subscribe to a last-by table and locally convert to a stream table, which they could then convert to an append or ring table to essentially keep client side history wrt the last-by table.

From a publishers perspective, a last-by table may often be preferable to an append, ring, or stream table.

**Comment by rcaudy** [comment]
This is necessarily a lossy conversion. You'll get a log of changes you've seen, not of all changes that occurred.

We do this conversion logically in the listener for Kafka exhaust, but manifesting as a stream table instead of a set of messages would not be hard.

Unlike the Kafka case, this could be reassembled to consistent full-table states without additional messaging, since the cycle is still part of propagation.

That said, I don't see an obvious use-case. Care to offer some, @devinrsmith ?

**Comment by devinrsmith** [comment]
Yep, it's explicitly lossy (just like last-by is). It's giving the client the ability to essentially turn a last-by table (which may be all the publisher wants to support) into a "historical" table with non-exhaustive updates over time.
